File: Assessment/Assessment.WEB/src/app/services/level.service.ts
import { Inject } from '@angular/core';
import { Http } from '@angular/http';
import { Level } from '../entities/level';

export class LevelService {
    http: Http;
    baseUrl: string;
    controller: string = "api/Level/";

    constructor(http: Http, baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    public getAll() {
        return this.http.get(this.baseUrl + this.controller);
    }

    public get(id: number) {
        return this.http.get(this.baseUrl + this.controller + id);
    }

    public update(level: Level) {
        return this.http.put(this.baseUrl + this.controller, level);
    }

    public insert(level: Level) {
        return this.http.post(this.baseUrl + this.controller, level);
    }

    public delete(id: number) {
        return this.http.delete(this.baseUrl + this.controller + id);
    }
}

File: Assessment/Assessment.Entities/LevelService.cs
using System;
using System.Collections.Generic;
using Assessment.Entities.DataInterface;

namespace Assessment.Entities
{
    public class LevelService
    {
        private ILevelRepository _repository;

        public LevelService(ILevelRepository repository)
        {
            _repository = repository;
        }

        public void Insert(Level level)
        {
            _repository.Insert(level);
        }

        public void Delete(int levelID)
        {
            _repository.Delete(levelID);
        }

        public void Update(Level level)
        {
            _repository.Update(level);
        }

        public Level Get(int levelID)
        {
            return _repository.Get(levelID);
        }

        public IEnumerable<Level> GetAll()
        {
            return _repository.GetAll();
        }
    }
}